### PR TITLE
27 integrate schema with gemini

### PR DESCRIPTION
**Pull request by absurdfarce**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1198513180644180/1198984111002425) by [Unito](https://www.unito.io/learn-more)

### CODE REVIEW

File: helm/adelphi/templates/schema-job.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ spec:
   templates:
   - name: configure-schema
     container:
-      image: alpine
+      image: frolvlad/alpine-python2
```
Comment (absurdfarce) [inline helm/adelphi/templates/schema-job.yaml:12]: Base alpine image didn't include Python

File: helm/adelphi/templates/schema-job.yaml [hunk 2]
```diff
@@ -20,6 +20,9 @@ spec:
           echo "Configuring schema on the target cluster...";
           cat /schema/schema.cql | kubectl exec -n {{ .Values.namespace }} -i -t -c cassandra {{ .Values.target.clusterName }}-{{ .Values.target.dc }}-default-sts-0 -- /opt/cassandra/bin/cqlsh -u {{ .Values.target.clusterName }}-superuser -p $(cat /secret-target/password);
           echo "Done";
+          pip install cassandra-driver;
+          wget -q https://raw.githubusercontent.com/datastax/adelphi/master/schema-anonymizer/anonymizer.py;
```
Comment (absurdfarce) [inline helm/adelphi/templates/schema-job.yaml:24]: This setup will also be removed once most of this installation logic is converted to the Docker images

File: helm/adelphi/values.yaml [hunk 3]
```diff
@@ -58,6 +58,7 @@ cql_schema: |-
     value text,
     PRIMARY KEY (key)
   );
+gemini_schema_keyspace: testkeyspace
```
Comment (absurdfarce) [inline helm/adelphi/values.yaml:61]: Anonymizer script only supports a single keyspace when generating Gemini output.  This val is used to constrain that operation.

File: helm/adelphi/templates/schema-job.yaml [hunk 4]
```diff
@@ -45,4 +45,4 @@ spec:
         secretName: {{ .Values.target.clusterName }}-superuser
         items:
         - key: password
-          path: password
\ No newline at end of file
+          path: password
```
Comment (absurdfarce) [inline helm/adelphi/templates/schema-job.yaml:48]: A side effect of an earlier impl which tried to perform schema extraction in the schema-job template

**Comment by absurdfarce** [comment]
Observed clean run of entire Argo workflow (including gemini-oracle-sut template) with this configuration

File: helm/adelphi/templates/gemini-job.yaml [hunk 5]
```diff
@@ -47,6 +48,21 @@ spec:
       volumeMounts:
       - name: workspace
         mountPath: /workspace
+    - name: extract-schema
+      image: python:3.10.0a1-alpine3.12
```
Comment (grighetto) [inline helm/adelphi/templates/gemini-job.yaml:52]: @absurdfarce fyi, I replaced the original image you had here with the official one for Python.

**Comment by grighetto** [comment]
@absurdfarce This works fine 👍 but I noticed it anonymizes the schema again when it runs the script to extract the Gemini schema. I think that's not desirable since we want to test the original schemas as they have been contributed to the project.
I'll open a bug ticket to fix that separately, it doesn't need to block this PR.

**Review by grighetto — APPROVED** [review]

**Pull request merged**